src/UserRepository/HtpasswdFactory.php

```diff
@@ -17,9 +17,7 @@ class HtpasswdFactory
      */
     public function __invoke(ContainerInterface $container) : Htpasswd
     {
-        $config = $container->has('config') ? $container->get('config') : [];
-        $htpasswd = $config['authentication']['htpasswd'] ?? null;
-
+        $htpasswd = $container->get('config')['authentication']['htpasswd'] ?? null;
```

Comment:
Why this particular change?

Reply:
Just because it's less line of code and we don't need `$config`.

Reply:
The previous version wouldn't fail immediately if the `config` service was not present, but, ultimately, it would fail anyways, so I'm okay with the change.

Reply:
@ezimuel @weierophinney wait... shouldn't container throw an exception (ContainerException) when service is not defined?
Please bare in mind we are mocking container always in tests, so not sure if it's right change.

Reply:
Ofc, check this out:

```php
    public function testContainer()
    {
        $container = new ServiceManager();
        $container->get('config');
    }
```

throws exception:
```
...Test::testContainer
Zend\ServiceManager\Exception\ServiceNotFoundException: Unable to resolve service "config" to a factory; are you certain you provided it during configuration?
```

Reply:
Right, and that's what I was getting at. The change that @ezimuel makes here will now emit that particular exception if the `config` service is not defined. It raises another exception if the service is defined, but the expected configuration is missing. That seems like a reasonable change, as it actually gets more precise about the cause of failure.

Reply:
@weierophinney So we are missing that test case in tests now. By default container is prophesized to return null on all methods, but in that case exception should be thrown.

Reply:
Okay, will add that test and push to @ezimuel 's branch shortly.